core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -951,15 +951,22 @@ class KafkaApis(val requestChannel: RequestChannel,
               new OffsetFetchResponse(Errors.NONE, authorizedPartitionData.asJava, header.apiVersion)
             }
           } else {
-            val (authorizedPartitions, unauthorizedPartitions) = offsetFetchRequest.partitions.asScala
-              .partition(authorizeTopicDescribe)
-            val (error, authorizedPartitionData) = coordinator.handleFetchOffsets(offsetFetchRequest.groupId,
-              Some(authorizedPartitions))
+            val (authorizedPartitions, unauthorizedPartitions) =
+              offsetFetchRequest.partitions.asScala.partition(authorizeTopicDescribe)
+            val (error, authorizedPartitionData) =
+              coordinator.handleFetchOffsets(offsetFetchRequest.groupId, Some(authorizedPartitions))
             if (error != Errors.NONE)
               offsetFetchRequest.getErrorResponse(error)
             else {
+              val authorizedAndValidatedPartitionData = authorizedPartitionData.map {
+                case (topicPartition, offsetInfo) =>
+                  if (metadataCache.getPartitionInfo(topicPartition.topic, topicPartition.partition).isEmpty)
```

Comment:
@vahidhashemian It seems to me that the difference also exists when some (but not all) partitions are unknown. Say the request contains two partitions and one of them is unknown, and the group is not available. I think we should return unknown for one partition and GROUP_COORDINATOR_NOT_AVAILABLE for the other partition. What do you think?